### PR TITLE
Spring-Security p2

### PR DESCRIPTION
**Pull request by yuriikozar**
Solution + Checkstyle
v1

### CODE REVIEW

**Review by kseniabykh — APPROVED** [review]
Nice job!

File: src/main/java/mate/academy/spring/dao/RoleDao.java [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package mate.academy.spring.dao;
+
+import java.util.Optional;
+import mate.academy.spring.model.Role;
+
+public interface RoleDao {
+
```
Comment (KseniiaMakarova) [inline src/main/java/mate/academy/spring/dao/RoleDao.java:7]: redundant empty line

File: src/main/java/mate/academy/spring/dao/impl/UserDaoImpl.java [hunk 2]
```diff
@@ -20,7 +20,8 @@ public UserDaoImpl(SessionFactory factory) {
     public Optional<User> findByEmail(String email) {
         try (Session session = factory.openSession()) {
             Query<User> findByEmail = session.createQuery(
-                    "FROM User WHERE email = :email", User.class);
+                    "FROM User u join fetch u.roles "
```
Comment (KseniiaMakarova) [inline src/main/java/mate/academy/spring/dao/impl/UserDaoImpl.java:23]:
```suggestion
                    "FROM User u JOIN FETCH u.roles "
```

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]
good job, left tiny comments

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request closed without merging**